Darksky enhancements

rdbahm opened the pull request:
## Description:
Made "Dew Point" capitalization in Darksky consistent with other two word sensors in Darksky, and added UV Index to supported sensors. All changes contained in darksky.py and no additional requirements.

**Related issue (if applicable):** N/A

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#3610

## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: darksky
    api_key: !secret api_key
    monitored_conditions:
      - uv_index
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

rdbahm committed:
Correct capitalization inconsistency in DarkSky

All two-word sensors ("Precip Intensity," "Nearest Storm Bearing," etc) in Darksky uses title case for the friendly name of the sensor, with the exception of "Dew point."

rdbahm committed:
Implement UV Index in Darksky

rdbahm committed:
Fixed whitespace for Tox compliance

homeassistant commented:
Hi @rdbahm, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9851).

Once you do that we will be able to review and accept this pull request.

Thanks!

rdbahm commented on homeassistant/components/sensor/darksky.py:
```diff
@@ -96,6 +96,9 @@
     'precip_intensity_max': ['Daily Max Precip Intensity',
                              'mm', 'in', 'mm', 'mm', 'mm', 'mdi:thermometer',
                              ['currently', 'hourly', 'daily']],
+    'uv_index': ['UV Index', None, None, None, None, None,
```
UV Index is a unitless measurement, but without a unit, it won't show up well in history graphs.

Is there a standard in HASS for unitless numeric measurement units?

armills replied:
I'd say that although there generally isn't a formal unit attached, UVI isn't a ratio/fraction/count or something else truly unit-less. It just is it's own unit. Hard to find some precedence, but in the [Wikipedia Description](https://en.wikipedia.org/wiki/Ultraviolet_index#Description), they do use the phrase +- 1 UV Index Unit.

We've got one precedent I could find searching the repo here: https://github.com/home-assistant/home-assistant/blob/46e030662da4ab7c653a23e1f34e527c9a0e7a8f/homeassistant/components/sensor/isy994.py#L80

If you could add that string to [const.py](https://github.com/home-assistant/home-assistant/blob/4cf300a7100edd61b7c558d3ef1cc33b5e1fc0cb/homeassistant/const.py#L250-L280), as `UV_INDEX` or `UNIT_UV_INDEX`, you can then import and use it here as the unit, and replace the usage in isy994. It's not perfect, but that will at least give us a formal precedent for how we handle it.

rdbahm committed:
Add unit for UV Index.

Per recommendation of reviewer, added 'UV Index' as a CONST in const.py, then used that const in both DarkSky and ISY994. It looks like BloomSky might also support UV Index and it should probably be standardized.

rdbahm commented:
Made a few changes based on review from @armills to add unit to UV Index sensor.

emlove reviewed (approved):
Looks great! Thanks for the contribution! 🎉
